### ISSUE TITLE
"508 heavily distorted objects" and shapenet preprocessing

### ISSUE THREAD

**Issue by normster** [jchibane/if-net#19]
Thank you for releasing and maintaining your code!

What's the difference between the 15 meshes to skip in "voxelization_32.npy" and the 508 heavily distorted objects mentioned in your paper?

Also, do you mind elaborating on the watertight mesh preprocessing in Wu and Wang et al. (DISN)? I wasn't able to find any details in their paper.

**Comment by jchibane** [comment]
Hi Norman,

you probably refer to this line in the readme:
> In order to remove meshes that could not be preprocessed (should not be more than around 15 meshes) you should run

I found the preprocessing failed sometimes (maybe due to concurrent read/write processes when processing in parallel or some defects in the data). In case the preprocessing could not succeed we need to remove them from our train/test.
Some further investigation on what the exact reason is could certainly be done here.

The 'heavily distorted' objects are objects that could not properly be made watertight by the algorithm from DISN.
We reused the data they released and also do not have further insides as given in their paper. However, I can suggest to use the watertighting procedure used by [Occupancy Networks](https://github.com/autonomousvision/occupancy_networks) (Lars Mescheder et al) - they had much less distortions or artifacts.

Best,
Julian
